components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -229,7 +230,8 @@ class GeckoEngineSession(
         val findFlags = if (forward) 0 else GeckoSession.FINDER_FIND_BACKWARDS
         geckoSession.finder.find(null, findFlags).then { result: GeckoSession.FinderResult? ->
             result?.let {
-                notifyObservers { onFindResult(it.current, it.total, true) }
+                val activeMatchOrdinal = if (it.current > 0) it.current - 1 else it.current
+                notifyObservers { onFindResult(activeMatchOrdinal, it.total, true) }
```

Comment:
What's the reason for rewriting those?

Reply:
[FinderResult.current ](https://dxr.mozilla.org/mozilla-central/source/mobile/android/geckoview/src/main/java/org/mozilla/geckoview/GeckoSession.java#1570)page starts from 1 while, [Webview version is zero based activeMatchOrdinal](https://developer.android.com/reference/android/webkit/WebView.FindListener#onFindResultReceived(int,%20int,%20boolean))

Also our  own data type is zero based. This could inconsistencies between engines 

  ```kotlin
  /**
     * A value type representing a result of a "find in page" operation.
     *
     * @property activeMatchOrdinal the zero-based ordinal of the currently selected match.
     * @property numberOfMatches the match count
     * @property isDoneCounting true if the find operation has completed, otherwise false.
     */
    data class FindResult(val activeMatchOrdinal: Int, val numberOfMatches: Int, val isDoneCounting: Boolean)
```

Reply:
Ah, makes sense. Would it be easy to adapt the WebView flavor? Would prefer we keep our API closer to GeckoView's. :)

Reply:
That sounds cool, but the issue is this will [break compatibility for FireFox lite](https://github.com/mozilla-tw/FirefoxLite/blob/b86fe9e05d5fe0cc9698441c236d8b23d2469d8d/app/src/main/java/org/mozilla/focus/widget/FindInPage.kt#L65) implementation.

Reply:
They do not use any engine component yet? :)

Reply:
They created their own widget using [org.mozilla.components:browser-session](https://github.com/mozilla-mobile/focus-android/blob/98a5700d6ed1c3d862e45da54911d94605450599/app/build.gradle#L191) and expecting that [FindResult.activeMatchOrdinal ](https://github.com/mozilla-mobile/android-components/blob/2ab783c5efffc272b22fa6336f98438f22a8e46c/components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt#L137) was zero-based.

  ```kotlin
  /**
     * A value type representing a result of a "find in page" operation.
     *
     * @property activeMatchOrdinal the zero-based ordinal of the currently selected match.
     * @property numberOfMatches the match count
     * @property isDoneCounting true if the find operation has completed, otherwise false.
     */
    data class FindResult(val activeMatchOrdinal: Int, val numberOfMatches: Int, val isDoneCounting: Boolean)
```